### DISCUSSION TITLE
Keep model loaded when using api

### DISCUSSION THREAD

**Discussion by nephi-dev** [Comfy-Org/ComfyUI#3103] [Q&A] [answered]
There's any way to keep a model loaded when using the api?

Example, on a first request the model "sdxl.safetensors" is loaded, but on the second request it got loaded again, slowing the api

**Comment by asagi4** [comment, marked as answer]
What kind of workflows are you sending to the API? In my experience the model stays loaded as long as the prompt you send in doesn't change in ways that require reloading.

Are you using the same node ID for the checkpoint loader between prompts?

**Reply by nephi-dev** [reply]
yep, same id for every node, yet the model keep re-loading

**Reply by asagi4** [reply]
Do you send a client id to the server? I think that may affect caching. You send it with the request when you connect to the websocket server and when you queue a prompt. If you don't send any, the server will just generate a random one.

**Reply by nephi-dev** [reply]
I'm not sending any, where do I send it?

**Reply by nephi-dev** [reply]
another thing @asagi4 I'm not using websocket, it's just the bare api

**Reply by asagi4** [reply]
You send it when you queue the prompt. I have code like this
```
def queue_prompt(prompt, client_id):
    p = {"prompt": prompt, "client_id": client_id}
    data = json.dumps(p).encode('utf-8')
    req =  urllib.request.Request("https://{}/prompt".format(server_address), data=data)
    return json.loads(urllib.request.urlopen(req).read())
```

Just generate the client id and keep it constant and that might help.

**Reply by nephi-dev** [reply]
figured out my problem, the ids was always adding up, got it fixed now, thanks for the help
